src/Tiqr/Legacy/TiqrUser.php

```diff
@@ -224,7 +224,15 @@ public function getNotificationType(): string
     public function getNotificationAddress(): string
     {
         try {
-            return $this->userStorage->getNotificationAddress($this->userId);
+            $notificationAddress=$this->userStorage->getNotificationAddress($this->userId);
+
+            // Filter bogus "null" and "(null)" entries that were put in the database by returning '' instead.
+            // There is no point in trying to send a notification to such addresses.
+            $notificationAddressLower = strtolower($notificationAddress);
+            if (($notificationAddressLower == 'null') || ($notificationAddressLower == '(null)')) {
```

Comment:
First off, this is a very effective way to stop this value from being used as the notification address! One would hope the Tiqr server user storage would prevent this from happening. 

Wouldn't it be better to let the source of the data guard the values being sent out? Its always a bit of a debate who is responsible for data integrity.

Finally. Have you actually seen a `(null)` string value? Or is this what you see in the database GUI when inspecting the field? I've never seen this happening before.

Reply:
`(null)` is how objective-C (used in the iOS app) converts null/nil values to string, so we see this one a lot.

The expectation of the server is that the notificationAddress is an opaque string, so that is why it did/does not enforce anything. This happening is unintended behaviour by the apps, and it has been happening in the past. We've been fixing this, but that does not change what is in the database currently. Also we're not willing to force people to update the app over this (which would mean locking them out if they don't) as it is merely an annoyance, not a security bug.

Yes, solving this in the userStorage would have been an option too. But the userStorage being an optional class, I did not want to change its interface, a bit academic, true. I actually "like" to see the wrong values appear in the data base, because this tells us when there is an issue how many accounts are affected. That is why I put the guard on the way out, not the way in.